api/envoy/extensions/filters/network/meta_protocol_proxy/matcher/v3/matcher.proto

```diff
@@ -0,0 +1,36 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.meta_protocol_proxy.matcher.v3;
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.meta_protocol_proxy.matcher.v3";
+option java_outer_classname = "MatcherProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Meta Protocol Proxy Route Matcher Configuration]
+
+// Used to match request service of the downstream request. Only relevant when the application
+// protocol is RPC such as Dubbo or Thrift.
+// [#not-implemented-hide:]
+message ServiceMatchInput {
```

Comment:
Are service and method just special properties that we think are common enough to have specific types? Or is there something else that makes them special?

I would probably have the comment be a bit more vague here if they are just special properties, like `Only applicable if a method provided by the application protocol`. gRPC for example is an RPC protocol that doesn't use methods (outside of google.api.annotations)

Reply:
@snowp your concern is legitimate, not every RPC protocol has these two properties.